src/FSqlMigrationManagerPostgres.ts

```diff
@@ -95,12 +153,61 @@ export class FSqlMigrationManagerPostgres extends FSqlMigrationManager {
 	}
 
 	protected async _removeVersionLog(executionContext: FExecutionContext, sqlConnection: FSqlConnection, version: string): Promise<void> {
+		await sqlConnection.statement(`
+			DELETE FROM "${this._schema}"."${this.versionTableName}_rollback_script"
+			WHERE "version_id" = (
+				SELECT "id"
+				FROM "${this._schema}"."${this.versionTableName}"
+				WHERE "version" = $1
+			)
+		`).execute(executionContext, version);
+	
 		await sqlConnection.statement(
 			`DELETE FROM "${this._schema}"."${this.versionTableName}" ` +
 			`WHERE "version" = $1`
 		).execute(executionContext, version);
 	}
 
+	protected async _getRollbackScripts(executionContext: FExecutionContext, sqlConnection: FSqlConnection, version: string): Promise<Array<FSqlMigrationSources.Script>> {
+		const scripts: Array<FSqlMigrationSources.Script> = [];
+
+		const sqlRecords: ReadonlyArray<FSqlResultRecord> = await sqlConnection
+			.statement(`
+				SELECT "id", "name", "kind", "file", "content"
+				FROM "__migration_rollback_script"
+				WHERE "version_id" = (SELECT "id" FROM "__migration" WHERE "version" = $1)
+			`)
+			.executeQuery(executionContext, version);
+
+		for (const sqlRecord of sqlRecords) {
+			const id: number = sqlRecord.get("id").asInteger;
+			const name: string = sqlRecord.get("name").asString;
+			const file: string = sqlRecord.get("file").asString;
+			const content: string = sqlRecord.get("content").asString;
+			const kindStr: string = sqlRecord.get("kind").asString;
+
+			let kind: FSqlMigrationSources.Script.Kind;
+			switch (kindStr) {
+				case FSqlMigrationSources.Script.Kind.JAVASCRIPT:
+				case FSqlMigrationSources.Script.Kind.SQL:
+				case FSqlMigrationSources.Script.Kind.UNKNOWN:
+					kind = kindStr;
+					break;
+				default:
+					throw new FExceptionInvalidOperation(`Cannot read a script (id:${id}) from database due not supported kind '${kindStr}'.`);
+			}
+
+			const script: FSqlMigrationSources.Script = new FSqlMigrationSources.Script(
+				name,
+				kind,
+				file,
+				content
+			);
+			scripts.push(script);
+		}
```

Comment:
Try to rewrite to using `sqlRecords.map(...)`

Reply:
Rewrited to using sqlRecords.map(...) -
https://github.com/freemework/sql.postgres/commit/65bf02388dfe03eb91735c6e66cf75772a8dd628